go/arrow/table.go

```diff
@@ -96,9 +96,9 @@ func NewColumn(field Field, chunks *Chunked) *Column {
 	}
 	col.data.Retain()
 
-	if !TypeEqual(col.data.DataType(), col.field.Type) {
+	if !StorageTypeEqual(col.data.DataType(), col.field.Type) {
```

Comment:
this change doesn't make any sense to me. `TypeEqual` should already end up checking that the storage types are equal by virtue of the extension type itself implementing the equals check.

Reply:
This wasn't required and is removed, it's back to `TypeEqual` now.